includes/admin/class-wc-payments-admin.php

```diff
@@ -734,6 +734,7 @@ public function enqueue_payments_scripts() {
 						'refundedAmount'        => $order->get_total_refunded(),
 						'canRefund'             => $this->wcpay_gateway->can_refund_order( $order ),
 						'chargeId'              => $this->order_service->get_charge_id_for_order( $order ),
+						'hasOpenAuthorization'  => $this->order_service->has_open_authorization( $order ),
```

Comment:
To follow the proposed approach we'd need to  build the url to cancel the order here, something like:
```suggestion
						'hasOpenAuthorization'  => $this->order_service->has_open_authorization( $order ),
						'cancelOrderUrl'        => wp_nonce_url( admin_url( 'admin-ajax.php?action=woocommerce_mark_order_status&status=cancelled&order_id=' . $order->get_id() ), 'woocommerce-mark-order-status' ),
						
```

Reply:
Thank you for investigating the alternatives @mgascam, a good learning for me 😀